### DISCUSSION TITLE
Stuck in DLSS Balance on startup (Settings don't get saved)

### DISCUSSION THREAD

**Discussion by rodefc** [praydog/REFramework#696] [General Questions]
Hello, please I need your help.

Even though I'm using your latest build from yesterday, settings are not being saved when restarting RE4 or RE8. I use DLSS Quality but it reverts back to Balanced everytime. 

I think this is the reason why I select 0x00000008 AA_MODE_REPLAY_MODE_ALL and LOD BIAS (DX) -0.500 in Nvidia Profile Inspector, to no avail. I still get mipmapping issues with reflections in water and metal objects (white dots). That config is for DLSS Quality but the game keeps starting up with Balanced.

**Comment by TommyCreo21** [comment]
Not sure why your settings don't save. That has been working fine for me since the update a couple of days ago. Which version of the DLSS plugin are you using? Try updating to the latest version [from here.](https://www.techpowerup.com/download/nvidia-dlss-dll/)

I recommend DLSS Tweaks, which can [found here.](https://www.nexusmods.com/witcher3/mods/7925?tab=files)
With it, you can change the resolution scale for all the quality presets. That way you can set the resolution of the default "Balanced" mode to be whatever you want.

Furthermore, the shimmering "white dots" in reflections can be fixed by disabling bloom. I don't think it has anything to do with mipmap bias.

**Comment by rodefc** [comment]
Hi Tommy! I so appreciate your reply.

Yes, I'm using the latest DLSS file from Techpowerup. Some people say that it's better to use an older one (2.X) because the latest brings some kind of ghosting and stuff. Although I haven't seen any ghosting in-game (maybe in the menues), do you recommend to stick to the latest DLSS anyway?

Could this not remembering settings thing be related to folder permissions? Like it doesn't allow other programs to create files or overwrite them. I tried to unselect "Read Only" but it reverts everytime.

Will give a try to DLSSTweaks. Do you know any tutorial for like making Balanced act as Quality?

**Comment by TommyCreo21** [comment]
I think I did see some ghosting at some point using newer DLLs, but it seems fixed now. Not sure what did it. So yeah, I think it should be fine.

Does RE Framework remember your other settings? Do you have a file called "re2_fw_config.txt" in the root folder of the game? That should contain all the settings. Mine has (among other things) these lines that control the upscaler:

> TemporalUpscaler_Enabled=true
> TemporalUpscaler_SharpnessAmount=0.000000
> TemporalUpscaler_SharpnessEnable=false
> TemporalUpscaler_UpscaleQuality=1
> TemporalUpscaler_UseNativeResolution=false

Regarding DLSS Tweaks, you just open the "dlsstweaks.ini"  and all settings are explained pretty thoroughly. Just set [DLSSQualityLevels] to Enable=true, and then set Balanced = 0.677777 or whatever your heart desires. I personally use 0.8 quite often, which gives great image quality.

**Comment by rodefc** [comment]
Yes! I do have that file. It contains everything but the TemporalUpscaler stuff. 

These are the settings I have:

FreeCam_SpeedModifierFast_V2=160
FreeCam_SpeedModifierSlow_V2=164
FreeCam_ToggleKey_V2=-1
GameObjectsDisplay_Enabled=false
Graphics_DisableGUI=false
Graphics_ForceRenderResToWindow=false
Graphics_UltrawideFOVMultiplier=0.500000
Graphics_UltrawideFix=false
Graphics_UltrawideFixFOV=true
Graphics_UltrawideFixVerticalFOV=true
ManualFlashlight_Enabled=false
ManualFlashlight_IgnorePowerOnZones=false
ManualFlashlight_Key_V2=70
RE8VR_AutoHideLowerBodyCutscenes=true
RE8VR_AutoHideUpperBodyCutscenes=true
RE8VR_HideArms=false
RE8VR_HideLowerBody=false
RE8VR_HideUpperBody=false
REFrameworkConfig_FontSize=16
REFrameworkConfig_MenuKey_V2=45
REFrameworkConfig_MenuOpen=true
REFrameworkConfig_RememberMenuState=true
Scene_TimeScale=1.000000
Scene_TimeScaleContinuousKey=-1
Scene_TimeScaleEnabled=false
Scene_TimeScaleToggleKey=-1
ScriptRunner_GarbageCollectionBudget=1000.000000
ScriptRunner_GarbageCollectionHandlerV2=0
ScriptRunner_GarbageCollectionMajorMultiplier=100.000000
ScriptRunner_GarbageCollectionMinorMultiplier=1.000000
ScriptRunner_GarbageCollectionMode=0
ScriptRunner_GarbageCollectionTypeV2=0
ScriptRunner_LogToDisk=false
VR_2DUIDistance=1.000000
VR_2DUIScale=12.000000
VR_AllowEngineOverlays=true
VR_AlternateFrameRendering=false
VR_CustomViewDistance=500.000000
VR_DecoupledPitch=false
VR_DesktopRecordingFix=true
VR_DesktopRecordingFixSkipPresent=true
VR_ForceAntiAliasing=true
VR_ForceDynamicShadows=true
VR_ForceFPS=true
VR_ForceLensDistortion=true
VR_ForceLensFlares=true
VR_ForceMotionBlur=true
VR_ForceVSync=true
VR_ForceVolumetrics=true
VR_HMDOrientedAudio=true
VR_JoystickDeadzone=0.150000
VR_MotionControlsInactivityTimer=30.000000
VR_OpenXRResolutionScale=1.000000
VR_RecenterViewKey=-1
VR_SetStandingOriginKey=-1
VR_UseCustomViewDistance=false
VR_WorldSpaceUIScale=15.000000

Bad luck, isn't it?

I clicked in the Reframework UI this:

REFrameworkConfig_RememberMenuState=true

And it got saved... so folder permissions are not the problem I guess.

Should I delete the file? What of this:

TemporalUpscaler_Enabled=true
TemporalUpscaler_SharpnessAmount=0.000000
TemporalUpscaler_SharpnessEnable=false
TemporalUpscaler_UpscaleQuality=1
TemporalUpscaler_UseNativeResolution=false

Means you have DLSS at "Quality" ? Maybe if I just copy these settings into the ini...?

**Comment by praydog** [comment]
You must not be using the latest version like you think. If your settings are getting saved, but not the TemporalUpscaler options, only reason I can think of is you did not correctly download and install the latest version.

https://github.com/praydog/REFramework/actions/runs/4534265122

The version on Nexus is also behaving as expected.

**Comment by rodefc** [comment]
I didn't know there was another version after the one from the 26th... I'll take a look!

By installing, you mean to just copy the dinput8.dll to the exe folder and that's it. It's well installed then hehe.

Btw, do you know if having these settings tweaked could make some performance hit?

«Antialiasing – Transparency Supersampling» al valor «0x00000008 AA_MODE_REPLAY_MODE_ALL».

«Texture Filtering – LOD Bias (DX) Quality: -0.5000 » 

If i put them back to default, could I get some FPS back? I see no difference to be honest. Before tweaking these settings the game already looked awesome.

**Comment by Gazlighter** [comment]
Can confirm this does indeed work now and settings save as intended as of the latest update. Thanks very much, performance and visuals through DLSS are on another level compared to the FSR implementation.

**Reply by rodefc** [reply]
Glad to hear that, but they still don't save for me, this is crazy!

**Comment by rodefc** [comment]
*UPDATE*

I deleted the ini file so reframe made a new one when I started up RE8. Now I have all lines in the file:

> Camera_DisableVignette=true
> Camera_Enabled=false
> Camera_FOV=81.000000
> Camera_FOVAiming=70.000000
> Camera_GlobalFOV=81.000000
> Camera_GlobalFOVEnabled=false
> Camera_VignetteBrightness=0.000000
> ChainViewer_Enabled=false
> FreeCam_DisableMoveKey_V2=-1
> FreeCam_DisableMovement=false
> FreeCam_Enabled=false
> FreeCam_LockKey_V2=-1
> FreeCam_LockPosition=false
> FreeCam_MoveDownKey_V2=162
> FreeCam_MoveUpKey_V2=32
> FreeCam_RotationSpeed=1.000000
> FreeCam_Speed=0.100000
> FreeCam_SpeedModifier=4.000000
> FreeCam_SpeedModifierFast_V2=160
> FreeCam_SpeedModifierSlow_V2=164
> FreeCam_ToggleKey_V2=-1
> GameObjectsDisplay_Enabled=false
> Graphics_DisableGUI=false
> Graphics_ForceRenderResToWindow=false
> Graphics_UltrawideFOVMultiplier=0.500000
> Graphics_UltrawideFix=false
> Graphics_UltrawideFixFOV=true
> Graphics_UltrawideFixVerticalFOV=true
> ManualFlashlight_Enabled=false
> ManualFlashlight_IgnorePowerOnZones=false
> ManualFlashlight_Key_V2=70
> RE8VR_AutoHideLowerBodyCutscenes=true
> RE8VR_AutoHideUpperBodyCutscenes=true
> RE8VR_HideArms=false
> RE8VR_HideLowerBody=false
> RE8VR_HideUpperBody=false
> REFrameworkConfig_FontSize=16
> REFrameworkConfig_MenuKey_V2=45
> REFrameworkConfig_MenuOpen=true
> REFrameworkConfig_RememberMenuState=false
> Scene_TimeScale=1.000000
> Scene_TimeScaleContinuousKey=-1
> Scene_TimeScaleEnabled=false
> Scene_TimeScaleToggleKey=-1
> ScriptRunner_GarbageCollectionBudget=1000.000000
> ScriptRunner_GarbageCollectionHandlerV2=0
> ScriptRunner_GarbageCollectionMajorMultiplier=100.000000
> ScriptRunner_GarbageCollectionMinorMultiplier=1.000000
> ScriptRunner_GarbageCollectionMode=0
> ScriptRunner_GarbageCollectionTypeV2=0
> ScriptRunner_LogToDisk=false
> **TemporalUpscaler_Enabled=true
> TemporalUpscaler_SharpnessAmount=0.000000
> TemporalUpscaler_SharpnessEnable=true
> TemporalUpscaler_UpscaleQuality=1
> TemporalUpscaler_UseNativeResolution=false**
> VR_2DUIDistance=1.000000
> VR_2DUIScale=12.000000
> VR_AllowEngineOverlays=true
> VR_CustomViewDistance=500.000000
> VR_DecoupledPitch=false
> VR_DesktopRecordingFix=true
> VR_DesktopRecordingFixSkipPresent=true
> VR_ForceAntiAliasing=true
> VR_ForceDynamicShadows=true
> VR_ForceFPS=true
> VR_ForceLensDistortion=true
> VR_ForceLensFlares=true
> VR_ForceMotionBlur=true
> VR_ForceVSync=true
> VR_ForceVolumetrics=true
> VR_HMDOrientedAudio=true
> VR_JoystickDeadzone=0.150000
> VR_MotionControlsInactivityTimer=30.000000
> VR_OpenXRResolutionScale=1.000000
> VR_RecenterViewKey=-1
> VR_RenderingTechnique=1
> VR_SetStandingOriginKey=-1
> VR_UseCustomViewDistance=false
> VR_WorldSpaceUIScale=15.000000

The thing is, it still does not save the Quality preset and settings when restarting the game. And it should, I have the latest DLL / DLSS / UpsclerBasePlugin

This is so annoying!

**Comment by praydog** [comment]
Are you closing the menu? That is the only way that settings will save. I mean the X button or pressing insert.

**Comment by praydog** [comment]
I was able to reproduce the bug. I'll work on a fix. Looks like it only happens if you close the menu with the "X" button. Works fine with the insert key to close the menu.

**Comment by praydog** [comment]
Should be fixed now. Please wait for it to compile in about 10-15 minutes. You can find the latest `pd-upscaler` builds here: https://github.com/praydog/REFramework/actions

Let me know if it's still not working.

**Comment by rodefc** [comment]
Thank you! Should I download the latest? You submitted two more since this morning:

![image](https://user-images.githubusercontent.com/128942481/228971979-4248b74b-7bed-4ef6-b28a-812f03e82d9e.png)

I enter the latest pd-upscaler, click the link and I download RE8 files (Or any RE)

![image](https://user-images.githubusercontent.com/128942481/228972364-ffc39eec-372d-4e50-bbc3-66fa69a9f220.png)

Then I just put the dinput8.dll file into the game folder:

![image](https://user-images.githubusercontent.com/128942481/228972693-802df0d1-80d2-41fe-b1ee-d1b63a83c817.png)

Do I need another file from the ZIP? I'm sure I'm installing this correctly. In the game folder I have DLL / DLSS / UpsclerBasePlugin, and that's it.

**Comment by Gazlighter** [comment]
While the main settings are getting saved such as DLSS quality, sharpness, DLAA etc, no matter what I do I am not able to get the "AllowEngineTAA" setting to save. It defaults to off but many people may want to enable this as it fixes most of the flashing, shimmering bugs in reflections. I can't get it to save to the enabled state. Any ideas? Seems very strange considering the rest of the settings save just fine.

**Reply by TommyCreo21** [reply]
The flashing and shimmering reflection artifacts can be removed entirely by disabling Bloom in the game settings. I would personally recommend that instead of enabling AllowEngineTAA. Even DLAA combined with TAA is really blurry.

Even though I'm actually a fairly big fan of bloom, when done well and in moderation, I think it's a worthwhile sacrifice to make for getting rid of the shimmering.

**Reply by Gazlighter** [reply]
Thanks, I'll give that a try.
